### ISSUE TITLE
Could someone share an example of multiclass segmentation?

### ISSUE THREAD

**Issue by zhongqiu1245** [qubvel/segmentation_models.pytorch#167]
Hi
I tried before but always failed in multi-class segmentation, the predicted mask is always empty. Could someone share an example of multi-class segmentation?
Thank you very much!

**Comment by SinclairHudson** [comment]
Hi! This is a hard problem to debug. I'm currently facing the same problem, but I think I've fixed it. You can see the core of my code here: https://app.wandb.ai/sinclairhudson/wato-roadline-segmentation/runs/26wj4o1k/code?workspace=user-sinclairhudson

Image segmentation is a hard task. This is a guess, but I suspect it's your model just not learning to predict anything other than background. Here are some steps I would take to debug:

- Double check your visualization. Try visualizing a label. It could be that you're just not visualizing the output properly. As you can see, I use Weights and Biases, which is a great analytics tool that helps me visualize all kinds of things.
- Weight your classes in your loss function. If your dataset is mostly background, your network will learn to only predict background because it's statistically the most probable correct answer. One way to combat this class imbalance is to weight your loss function; give all your other classes a higher weight so the network pays more attention to them and learns to predict them correctly.
- Check your training parameters. Your model could have diverged, it could be not learning at all. Try replacing your big complicated network with a dummy example to make sure everything else is working correctly. It's handy to visualize the weights of your network to make sure they're actually changing, ie the network is actually learning.
- Train for longer. These networks take a long time to train, especially from random initialization. It could be that the network hasn't had the time to learn anything and is just predicting background until it learns small features that make it more confident in different (correct) predictions.

**Comment by pvthuy** [comment]
@zhongqiu1245 Did you check this? https://github.com/qubvel/segmentation_models.pytorch/issues/13

**Comment by gireeshkbogu** [comment]
The link won't work

**Comment by SinclairHudson** [comment]
@gireeshkbogu I've updated it, so it should go. I think I deleted the old run... It's a link to a project I'm still actively working on.

**Comment by zhongqiu1245** [comment]
thank you!@SinclairHudson

**Comment by marcomameli1992** [comment]
I'm trying to see your code but I do not find it. How can I use multiclass segmentation with this package and with weighting the classes?

**Comment by shirokawakita** [comment]
I upload an example of multiclass segmentation with segmentation-models. please enjoy it!
https://github.com/shirokawakita/multiclass-segmentation/blob/main/example_camvid_multiclassB_quita.ipynb

**Comment by zhongqiu1245** [comment]
thank you@shirokawakita
